### PR TITLE
feat: update error handle

### PR DESCRIPTION
**Pull request by jiacai2050**
Close <https://github.com/CeresDB/ceresdb/issues/138>

### CODE REVIEW

File: docs/src/en/dev/style_guide.md [hunk 1]
```diff
@@ -252,6 +239,34 @@ close_writer.context(WritingError {
...
+
+// Error in module B
+pub enum Error {
+    #[snafu(display("Another error, source:{}.\nBacktrace:\n{}", source, backtrace))]
+    AnotherError {
+        /// This error wraps another error that already has a
+        /// backtrace. Instead of capturing our own, we forward the
+        /// request for the backtrace to the inner error. This gives a
+        /// more accurate backtrace.
+        #[snafu(backtrace)]
```
Comment (ShiKaiWi) [inline docs/src/en/dev/style_guide.md:264]: Is this necessary? It seems it is never used in current codebase.

**Review by Rachelint — APPROVED** [review]
LGTM

File: docs/src/en/dev/style_guide.md (hunk 1 above)
Comment (Rachelint) [reply docs/src/en/dev/style_guide.md:264]: It sounds good to do like that when impl new features?

Comment (jiacai2050) [reply docs/src/en/dev/style_guide.md:264]: It's optional, this is what snafu docs recommend. I think we should follow it.

It we don't add it, then `anotherError.getBacktrace()` would return None, we have to call `anotherError.source.getBacktrace()` to get underlying backtrace.

Comment (jiacai2050) [reply docs/src/en/dev/style_guide.md:264]: https://docs.rs/snafu/latest/src/snafu/guide/examples/backtrace.rs.html#81

**Pull request merged**